presto-parquet/src/main/java/io/prestosql/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -260,6 +310,59 @@ public static Domain getDomain(
         return Domain.create(ValueSet.all(type), hasNullValue);
     }
 
+    @VisibleForTesting
+    public Domain getDomain(Type type, long rowCount, ColumnIndex columnIndex, ParquetDataSourceId id, RichColumnDescriptor descriptor)
+            throws ParquetCorruptionException
+    {
+        if (columnIndex == null) {
+            return Domain.all(type);
+        }
+
+        String columnName = descriptor.getPrimitiveType().getName();
+
+        if (isCorruptedColumnIndex(columnIndex)) {
+            throw corruptionException(columnName, id, columnIndex);
+        }
+
+        if (isEmptyColumnIndex(columnIndex)) {
+            return Domain.all(type);
+        }
+
+        long totalNullCount = columnIndex.getNullCounts().stream().reduce(0L, (a, b) -> a + b);
+        if (totalNullCount == rowCount) {
+            return Domain.onlyNull(type);
+        }
+
+        boolean hasNullValue = totalNullCount > 0;
+
+        if (descriptor.getType().equals(PrimitiveTypeName.BOOLEAN)) {
+            // After row-group filtering for boolean, page filtering shouldn't do more
+            return Domain.all(type);
+        }
+
+        if (descriptor.getType().equals(PrimitiveTypeName.INT32) || descriptor.getType().equals(PrimitiveTypeName.INT64) || descriptor.getType().equals(PrimitiveTypeName.FLOAT)) {
```

Comment:
Also, there's some inconsistency (and potential discrepancy) between this, which switches on the parquet type, vs the code in ColumnIndexValueConverter, which switches on the Trino type. I think we may have the wrong abstraction for the conversions -- let me look more deeply.

Reply:
For FLOAT, the code uses Float.floatToRawIntBits in ColumnIndexValueConverter.

Reply:
That would be a problem, because the comparison as float values is not equivalent with the comparison of values after they are converted to their IEEE754 representation and interpreted as ints:

```
jshell> float a = -1E-10f, b = -1E20f; a < b; Float.floatToRawIntBits(a) < Float.floatToRawIntBits(b);
a ==> -1.0E-10
b ==> -1.0E20
$68 ==> false
$69 ==> true
```

Also, there needs to be special handling for NaN.

Reply:
Definitely worthy of a test, but I believe Float.intBitsToFloat is used to convert back before comparisons are done.

Reply:
That doesn't seem to be the case. `createDomain` just creates ranges using the raw values (as long).

Reply:
Will add getMin/MaxValuesAsFloat . Or perhaps the getMin/MaxValuesAsDouble can be used in this case. Would be nice if the Type could be used instead of all these if statements. Do you think that's feasible?

Reply:
Yes, possibly. See my other comment above about the potential inconsistencies between these checks and the ones in the converter, which are based on Type. I'll dig into this further as I continue reviewing, but it does seem like there's an opportunity for improving it.

Reply:
This mechanism of converting a Float to int using `floatToRawIntBits` is already in use with statistics [here](https://github.com/trinodb/trino/blob/577302b514687331c9996fb8623b62dbf8c71b29/lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java#L207). Do you want me to change it there too?

Reply:
Filed https://github.com/trinodb/trino/issues/8893 to improve this.